test/core/templaters/jinja_test.py

```diff
@@ -507,6 +507,23 @@ def test__templater_full(subpath, code_only, include_meta, yaml_loader, caplog):
                 (" BAR", "literal", 30),
             ],
         ),
+        (
+            """{% set my_query %}
+select 1 from foobarfoobarfoobarfoobar_{{ "dev" }}
```

Comment:
Note the `{{ }}` block on this line inside the `{% set %}` block. This was previously not handled correctly and caused a runtime error.

Reply:
Might be good to link to the docs on this, as I didn't realise this was valid syntax, but looks like it is: https://jinja.palletsprojects.com/en/3.0.x/templates/#block-assignments

Reply:
Ah ignore me. See you added that comment in the code itself on line 314.

Reply:
Done!

Reply:
No worries, it's often a good idea to document both the tests and the code, to help connect the dots.